src/main/java/mat/client/shared/MeasureLibraryResultTable.java

```diff
@@ -104,7 +104,7 @@ public void run() {
                         }
                     };
                     singleClickTimer.schedule(MOUSE_CLICK_DELAY);
-                } else if (obj.getClickCount() == 2) {
+                } else if (obj.getClickCount() == 2 && obj.isFhirEditableFeatureFlag()) {
```

Comment:
1. My concern that the notion of feature-flags penetrates all aspects of the business logic...It makes our UI and DTO objects coupled with both FLAGS and FHIR (https://enterprisecraftsmanship.com/posts/cohesion-coupling-difference/).
I would suggest using 'isEditable'  instead. Then 'isEditable' / isVieweable can be calculated based on flags and model type. So we will minimize future code changes..

See https://enterprisecraftsmanship.com/posts/cohesion-coupling-difference/
2. Some of the items cannot be edited but can be viewed in read-only mode. The same handlers are used to view items. Current logic can possibly block it. Could you please double check it's not blocked?

Reply:
At first I tried to use existing 'isEditable' to do calculations based on flags and model type. But Mat.Context.get().getFeatureFlagStatus() isn't available in server side. When I tried to access MatContext in isCurrentMeasureEditable(). It is throwing null pointer.

Reply:
Well, feature flags are accessible through FeatureFlagService on the server-side.
Also, we might want to add caching (spring cache or hibernate caching) and optionally feature flag facade.